Core/Core/Planner/CalendarFilter/Model/UseCase/GetCalendarFilters.swift

```diff
@@ -74,10 +74,15 @@ class GetCalendarFilters: UseCase {
 
         Publishers
             .CombineLatest(coursesFetch, groupsFetch)
-            .map {
-                APIResponse(
-                    courses: $0.0,
-                    groups: $0.1.body
+            .map { (courses, groups) in
+                let validCourseIDs = courses.map { $0.id }
+                /// If a course's end date has passed and "Restrict students from viewing course after course end date" is checked
+                /// then fetching events for a group in this course will give 403 unauthorized.
+                /// To be on the safe side we drop all courses without a valid course.
```

Comment:
I guess it should be "drop all groups without"

Reply:
Agree, renamed.